app/services/repository_datatable_service.rb

```diff
@@ -0,0 +1,198 @@
+class RepositoryDatatableService
+
+  attr_reader :repository_rows, :assigned_rows, :mappings
+
+  def initialize(repository, params, user, my_module = nil)
+    @repository = repository
+    @user = user
+    @my_module = my_module
+    @params = params
+    create_columns_mappings
+    process_query
+  end
+
+  private
+
+  def create_columns_mappings
+    # Make mappings of custom columns, so we have same id for every
+    # column
+    i = 5
+    @mappings = {}
+    @repository.repository_columns.order(:id).each do |column|
+      @mappings[column.id] = i.to_s
+      i += 1
+    end
+  end
+
+  def process_query
+    order_obj = build_conditions(@params)[:order_by_column]
+    search_value = build_conditions(@params)[:search_value]
+    if search_value.present?
+      @repository_rows = sort_rows(order_obj, search(search_value))
+    else
+      @repository_rows = sort_rows(order_obj, fetch_records)
+    end
+  end
+
+  def fetch_records
+    repository_rows = RepositoryRow.preload(:repository_columns,
+                                            :created_by,
+                                            repository_cells: :value)
+                                   .joins(:created_by)
+                                   .where(repository: @repository)
+    if @my_module
+      @assigned_rows = @my_module.repository_rows
+                                 .preload(
+                                   :repository_columns,
+                                   :created_by,
+                                   repository_cells: :value
+                                 )
+                                 .joins(:created_by)
+                                 .where(repository: @repository)
+      return @assigned_rows if @params[:assigned] == 'assigned'
+    else
+      @assigned_rows = repository_rows.joins(
```

Comment:
I think we can use here simply '@assigned_rows = repository_rows.joins(:my_module_repository_rows)'